gap/base/recognition.gi

```diff
@@ -1017,10 +1048,52 @@ RECOG.TestGroup := function(g,proj,size)
       s := SLPforElement(ri,x);
       if s = fail or not(isequal(ri)(ResultOfStraightLineProgram(s,l),x)) then
           Print("ALARM: set count to -1 to skip test!\n");
+          Print("group := ", g, ";\n");
+          Print("recogsize := ", Size(ri), ";\n");
+          Print("proj := ", proj, ";\n");
+          Print("x := ", x, ";\n");
+          Print("s := ", s, ";\n");
+          if s <> fail then
+            Print("result := ", ResultOfStraightLineProgram(s,l), ";\n");
+          fi;
           Error("Alarm: SLPforElement did not work!\n");
+          
           if count = -1 then return fail; fi;
       fi;
-  until count >= 30;
+  until count >= options.inTests;
+
+  if IsPermGroup(g) then
+    supergroup := SymmetricGroup(LargestMovedPoint(g) + 2);
+  elif IsMatrixGroup(g) then
```

Comment:
Should there be a small comment here as to why we don't need `PGL(...) ` as a supergroup? 

If I understand this correctly this works because both a node for non-projective groups and a node for projective groups accept plain matrices (e.g. an `Is8BitRepMatrix`) .

Reply:
Longer term we should discuss what the 'supergroup' here should be. As Frank demonstrated, in GAP you can normally ask questions like `"cheese" in SymmetricGroup(3)` (which returns false), so maybe that should also be allowed here (at the moment it breaks, so I thought I'd start by only checking "sensible" things).

Reply:
Given an inner recognition node with a reduction  homomorphism I think there always is a maximal subgroup of (P)GL that admits this homonorphism. We could generate random elements from this group and send them through the subtree rooted in the current node. 

E.g. if our group is reducible, then after a base change it lies in a group of block triangular matrices. The corresponding reduction homomorphism is a projection onto one of the diagonal blocks.

Reply:
Can we already access this maximal subgroup from a recognition node? If not this could be one further issue.

Reply:
I don't understand what you mean with "this maximal subgroup"; in particular, when wouldn't it be all of GL resp. PGL? Anyway, I don't think this is relevant for merging this PR, is it?

Reply:
Oh, it's not relevant for this PR.  I'll have to open a separate issue for this and write the following thoughts down cleanly:

I was thinking about a way how to construct in our setting an overgroup of `g` which is "close" to `g`, as @ChrisJefferson put it. If we take a random elements of GL we test whether a reduction homomorphism correctly decides whether it can be applied to the random element. I think we (almost) never test the case where we generate a random element that we can apply the homomorphism to but which is not an element of `g`.

Reply:
I would be happy to see a bunch of extensions to this which generate "interesting" elements (either in, or not in, the group) which we think are likely to cause problems.